pkg/gardenlet/controller/seed/seed_care_control.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package seed
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	confighelper "github.com/gardener/gardener/pkg/gardenlet/apis/config/helper"
+
+	"github.com/sirupsen/logrus"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NewCareReconciler returns an implementation of reconcile.Reconciler which is dedicated to execute care operations
+// on shoots, e.g., health checks or garbage collection.
+func NewCareReconciler(
+	clientMap clientmap.ClientMap,
+	l logrus.FieldLogger,
+	config *config.GardenletConfiguration,
+) reconcile.Reconciler {
+	return &careReconciler{
+		clientMap: clientMap,
+		logger:    l,
+		config:    config,
+	}
+}
+
+type careReconciler struct {
+	logger    logrus.FieldLogger
+	clientMap clientmap.ClientMap
+	config    *config.GardenletConfiguration
+}
+
+func (r *careReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.logger.WithField("seed", req.String())
+
+	gardenClient, err := r.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	seed := &gardencorev1beta1.Seed{}
+	if err := gardenClient.Client().Get(ctx, req.NamespacedName, seed); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("[SEED CARE] Stopping care operations for Seed since it has been deleted")
```

Comment:
Please check whether you respect the new logging guidelines (https://github.com/gardener/gardener/blob/master/docs/development/logging.md)